src/net/bootstrapnodeupdater.cpp

```diff
@@ -0,0 +1,152 @@
+#include "bootstrapnodeupdater.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QRegularExpression>
+
+namespace {
+    const QUrl NodeListAddress{"https://nodes.tox.chat/json"};
+    const QLatin1Literal jsonNodeArrayName{"nodes"};
+    const QLatin1Literal emptyAddress{"-"};
+    const QRegularExpression ToxPkRegEx(QString("(^|\\s)[A-Fa-f0-9]{%1}($|\\s)").arg(64));
+}
+
+namespace NodeFields {
+    const QLatin1Literal status_udp{"status_udp"};
+    const QLatin1Literal status_tcp{"status_tcp"};
+    const QLatin1Literal ipv4{"ipv4"};
+    const QLatin1Literal ipv6{"ipv6"};
+    const QLatin1Literal public_key{"public_key"};
+    const QLatin1Literal port{"port"};
+    const QLatin1Literal maintainer{"maintainer"};
+    // TODO(sudden6): make use of this field once we differentiate between TCP nodes, and bootstrap nodes
+    const QLatin1Literal tcp_ports{"tcp_ports"};
+    const QStringList neededFields{status_udp, status_tcp, ipv4, ipv6, public_key, port, maintainer};
+}
+
+/**
+ * @brief Fetches a list of currently online bootstrap nodes from node.tox.chat
+ * @param proxy Proxy to use for the lookup, must outlive this object
+ */
+BootstrapNodeUpdater::BootstrapNodeUpdater(const QNetworkProxy &proxy, QObject* parent)
+    : QObject{parent}
+    , proxy{proxy}
+{
+}
+
+bool BootstrapNodeUpdater::requestBootstrapNodes()
+{
+    nam.setProxy(proxy);
+    connect(&nam, &QNetworkAccessManager::finished, this, &BootstrapNodeUpdater::onRequestComplete);
+
+    QNetworkRequest request{NodeListAddress};
+    request.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
+
+    nam.get(request);
+    return true;
+}
+
+void BootstrapNodeUpdater::onRequestComplete(QNetworkReply* reply)
+{
+    if(reply->error() != QNetworkReply::NoError){
```

Comment:
Add space after `if` and `for` for all PR

Reply:
Done